src/main/java/io/jenkins/plugins/coverage/source/DefaultSourceFileResolver.java

```diff
@@ -115,9 +119,19 @@ public void resolveSourceFiles(Run<?, ?> run, FilePath workspace, TaskListener l
                 } else {
                     listener.getLogger().printf("Cannot found source file for %s%n", sourceFilePath);
                 }
-            });
-        });
+            } finally {
+                // ensure latch will count down
+                latch.countDown();
+            }
+        }));
+
 
+        try {
+            latch.await(1, TimeUnit.HOURS);
```

Comment:
Maybe add to the list of future improvements to make this wait time a user configurable setting.

Reply:
Good idea, and will add it in the future.